### PR TITLE
#159952590 Build Authentication for Protected Endpoints with JWT

### PR DESCRIPTION
**Pull request by anu-001**
#### What does this PR do?
Have POST/ method for signup and login Rest Endpoints for Signup and Login with JWT

#### Description of Task to be completed?
Have the following endpoints working
* POST  /api/v2/auth/signup
* POST  /api/v2/auth/login

#### How should this be manually tested?
Clone the repoisitory and change directory in to it, run npm install
* Using Postman test every endpoints above with this header
key: Content-Type: value : application/x-www-form-urlencoded
* To test POST  /api/v2/auth/signup, use the body form, with username, email, and password with your choice words
* To test POST  /api/v2/auth/login, supply the email and password you used above, to generate a token
          
#### Any background context you want to provide?
This implementation do not have protected endpoints for now to test the jwt token

#### What are the relevant pivotal tracker stories?
[#159952590 ](https://www.pivotaltracker.com/story/show/159952590)
[#159952646](https://www.pivotaltracker.com/story/show/159952646)

### CODE REVIEW

**Comment by Kenec** [comment]
The title of your **PR**, **What does this PR do?**, and **How should this be manually tested?** does no tally. The title of your PR should reflect what your PR does. Also, you should list steps on how this PR can be tested, not a high-level description. Use this as a [sample](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices)

File: .env [hunk 1]
```diff
@@ -1,2 +1,7 @@
+ELEPHANTSQL_URL=postgres://noypnhlg:Mr8UvtnUw82xrp_09sC9sXSJ66qewV4a@baasu.db.elephantsql.com:5432/noypnhlg
```
Comment (Kenec) [inline .env]: Please remove the .env file from Github. It contains your application secrets and should never be pushed to the Git

File: api/v1/routes/index.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ const { check } = require('express-validator/check');
```
Comment (Kenec) [inline api/v1/routes/index.js:7]: - Remove "babel-polyfill". All your transpiling should be done in one place which is your script.
- All form of validation should be abstracted to a different file. 
-

File: api/v2/auth/authController.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import jwt from 'jsonwebtoken';
```
Comment (Kenec) [inline api/v2/auth/authController.js:1]: - Each line of code should not exceed 80 lines. use eslint to ensure this
- Remove all inline comment. Only JSDoc is required
- In your JSDoc, your specified response and request as parameters while the method has req and res. The JSDoc parameter should match that of the method.
- The status code of 404 is not appropriate for a record that already exists. Go through this [material](https://httpstatuses.com/)

File: api/v2/controllers/questionController.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import pool from '../config/config';
+/** for database testing */
```
Comment (Kenec) [inline api/v2/controllers/questionController.js]: - Remove all inline comments. Use JSDoc instead.

File: api/v2/middlewares/helper.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+/* This a helper class to get one of the question from the collection with id,
```
Comment (Kenec) [inline api/v2/middlewares/helper.js]: - Remove inline comments. Use JSDoc instead

**Review by Kenec — CHANGES_REQUESTED** [review]

File: api/v1/routes/index.js (hunk 2 above)
Comment (Anuoluwa) [reply api/v1/routes/index.js]: Noted, I will update it after merging.

**Pull request merged**